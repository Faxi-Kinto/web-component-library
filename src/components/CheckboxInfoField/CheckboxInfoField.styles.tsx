import styled from 'styled-components';
import { pxToRem } from '../../styles/basics';
import { theme } from '../../config';
import { padding } from '../../styles/basics';

export const Container = styled.div`
  line-height: ${pxToRem('20px')};
  border-bottom: ${pxToRem('1px')} solid ${theme.pallet.ACCENT_1};
  margin-bottom: ${pxToRem('42px')};

  .checkbox-info-field {
    &__error-msg {
      padding-right: ${pxToRem('50px')};
      font-size: ${pxToRem('14px')};
      color: ${theme.pallet.ERROR};
    }
    &__selected {
      color: ${theme.pallet.LIGHT_BLUE_2};
    }
    &__privacy-title {
      margin-bottom: ${pxToRem('30px')};
      font-weight: 500;
      max-width: ${pxToRem('315px')};
    }
    &__privacy-body {
      margin-bottom: ${pxToRem('42px')};
      display: block;
      ${padding('0', pxToRem('28px'), '0', '0')};
      font-size: ${pxToRem('14px')};
    }
  }
  &.checkbox-info-field__no-border {
    border-bottom: none;
  }
`;
