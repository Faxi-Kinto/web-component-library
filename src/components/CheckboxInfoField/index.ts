import CheckboxInfoField from './CheckboxInfoField.component';

export default CheckboxInfoField;
