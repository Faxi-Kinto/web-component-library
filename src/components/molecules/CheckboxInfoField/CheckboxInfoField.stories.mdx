import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';

import CheckboxInfoField from './';

<Meta title="Core|Molecules/CheckboxInfoField" />

# CheckboxInfoField Component

This field is a wrapper around [Checkbox](/docs/core-atoms-checkbox--default-story) atom and it is intended for use in [Form Field](/docs/core-molecules-formfield--default-story)
as part of "Terms and Conditions" form enabeling: error messages, error state for input atom and description message for the field.
`CheckboxInfoField` implements FormField API and it has minor styling/functionality changes in relation to `CheckboxField`

`label` and `description` are mandatory.

`lastElement` propery options out the bottom border line

#### Default

<Preview>
  <Story name="default" decorators={[withKnobs]}>
    <CheckboxInfoField
      label="Sign up to hear about the latest offers from Faxi and selected third parties"
      error={text('Error message', 'This field is required')}
      description={text('Description message', 'Faxi would like to contact you via in-app messages, push messages and email to promote other Faxi services and the services of selected third parties which we believe may be of interest to you. Please see our Privacy Policy for more details.')}
      labelPosition={'left'}
      lastElement={boolean('Last element (hide border)', false)}
      touched={boolean('Touched', false)}
      dirty={boolean('Dirty', false)}
    />
  </Story>
</Preview>

## Props

<Props of={CheckboxInfoField} />
