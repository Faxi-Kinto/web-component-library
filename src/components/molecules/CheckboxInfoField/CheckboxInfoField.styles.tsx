import styled from 'styled-components';
import { pxToRem } from 'styles/fonts';
import { theme } from 'config';

export const Container = styled.div`
  line-height: ${pxToRem('20px')};
  border-bottom: 1px solid ${theme.pallet.ACCENT_1};
  padding-bottom: 40px;
  .checkbox-info-field {
    &__error-msg {
      padding-right: 50px;
      font-size: ${pxToRem('14px')};
      color: ${theme.pallet.ERROR};
    }
    &__selected {
      color: ${theme.pallet.LIGHT_BLUE_2};
    }
    &__privacy-title {
      margin-bottom: 30px;
      font-weight: 500;
      max-width: 315px;
    }
    &__privacy-body {
      margin-bottom: 20px;
      display: block;
      padding: 0 28px 0 0;
      font-size: ${pxToRem('14px')};
    }
  }
  &.checkbox-info-field__no-border {
    border-bottom: none;
  }
`;
