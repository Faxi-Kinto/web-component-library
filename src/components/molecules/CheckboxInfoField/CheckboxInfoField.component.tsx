import React, { useState } from 'react';
import * as Styled from './CheckboxInfoField.styles';
import Checkbox from '../Checkbox';
import { FieldProps } from '../FormField/FieldProps';
import useField from 'hooks/useField';
import { CheckboxProps } from '../Checkbox/Checkbox.component';
import { Text } from 'components';

/**
 * @name CheckboxInfoField
 *
 * [Insert omponent description]
 *
 * @returns {JSX}
 */

export type CheckboxInfoFieldProps = FieldProps<boolean> & {
  label: string;
  description: string | JSX.Element | JSX.Element[];
  lastElement?: boolean;
} & CheckboxProps;

const CheckboxInfoField: React.FC<CheckboxInfoFieldProps> = (
  props: CheckboxInfoFieldProps
): JSX.Element => {
  const {
    value,
    onChange,
    error,
    description,
    label,
    lastElement,
    ...rest
  } = props;

  const [selected, setSelected] = useState(false);
  const showError = useField(props);
  return (
    <Styled.Container
      className={`checkbox-info-field ${lastElement &&
        'checkbox-info-field__no-border'}`}
    >
      <Checkbox
        {...rest}
        label={
          <Text.Body
            className={`checkbox-info-field__privacy-title ${selected &&
              'checkbox-info-field__selected'}`}
          >
            {label}
          </Text.Body>
        }
        checked={value}
        onChange={ev => {
          setSelected(!selected);
          onChange && onChange(ev.target.checked);
        }}
        error={showError}
      />
      <Text.Body className="checkbox-info-field__privacy-body">
        {description}
      </Text.Body>
      {showError ? (
        <div className="checkbox-info-field__error-msg">{error}</div>
      ) : (
        ''
      )}
    </Styled.Container>
  );
};

CheckboxInfoField.defaultProps = {};

export default CheckboxInfoField;
