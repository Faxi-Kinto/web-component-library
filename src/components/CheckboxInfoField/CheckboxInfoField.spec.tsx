import React from 'react';
import renderer from 'react-test-renderer';

import CheckboxInfoField from './CheckboxInfoField.component';

test('CheckboxField renders correctly', () => {
  const component = renderer.create(
    <CheckboxInfoField label="Hello" description="Test" />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
