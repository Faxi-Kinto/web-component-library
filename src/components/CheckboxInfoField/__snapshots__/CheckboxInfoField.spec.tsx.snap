// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField renders correctly 1`] = `
.c0 {
  line-height: 1.25rem;
  border-bottom: 0.0625rem solid #4E606A;
  margin-bottom: 2.625rem;
}

.c0 .checkbox-info-field__error-msg {
  padding-right: 3.125rem;
  font-size: 0.875rem;
  color: #EB4820;
}

.c0 .checkbox-info-field__selected {
  color: #00a5b5;
}

.c0 .checkbox-info-field__privacy-title {
  margin-bottom: 1.875rem;
  font-weight: 500;
  max-width: 19.6875rem;
}

.c0 .checkbox-info-field__privacy-body {
  margin-bottom: 2.625rem;
  display: block;
  padding: 0 1.75rem 0 0;
  font-size: 0.875rem;
}

.c0.checkbox-info-field__no-border {
  border-bottom: none;
}

.c1 {
  position: relative;
  cursor: pointer;
  min-height: 1.25rem;
  padding-left: 1.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > input {
  display: none;
}

.c1:before {
  content: '';
  display: inline-block;
  box-sizing: border-box;
  border: #4E606A solid 0.0625rem;
  position: absolute;
  top: 0;
  left: 0;
  width: 1.25rem;
  height: 1.25rem;
}

.c1 > span {
  line-height: 1.25rem;
}

.c1.checked:after {
  background: url('assets/icons/checkmark.svg') 100% no-repeat;
  content: '';
  width: 0.8125rem;
  height: 0.5625rem;
  position: absolute;
  top: 0.34375rem;
  left: 0.1875rem;
}

.c1.label--has-label-text {
  padding-left: 1.75rem;
}

.c1.label--has-error > input {
  color: #EB4820;
}

.c1.label--has-error:before {
  border: #EB4820 solid 0.0625rem;
}

.c1.label--left {
  padding: 0 1.75rem 0 0;
}

.c1.label--left:before {
  position: absolute;
  top: 0;
  right: 0;
  left: auto;
}

.c1.label--left.checked:after {
  position: absolute;
  top: 0.34375rem;
  right: 0.25rem;
  left: auto;
}

.c2 {
  padding: 0;
  margin: 0;
  font-size: 1rem;
  color: #4E606A;
  font-weight: 300;
  white-space: pre-wrap;
}

<div
  className="c0 checkbox-info-field undefined"
>
  <label
    className="c1 label--has-label-text "
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="c2 checkbox-info-field__privacy-title false"
      color="#4E606A"
      fontSize="16px"
    >
      Hello
    </span>
  </label>
  <span
    className="c2 checkbox-info-field__privacy-body"
    color="#4E606A"
    fontSize="16px"
  >
    Test
  </span>
  
</div>
`;
